src/sentry/testutils/cases.py

```diff
@@ -1232,6 +1235,122 @@ def _send_buckets(cls, buckets, entity):
+        hours_before_now: int = 0,
+        minutes_before_now: int = 0,
+        seconds_before_now: int = 0,
+    ):
+        # We subtract one second in order to account for right non-inclusivity in the query. If we wouldn't do this
+        # some data won't be returned (this applies only if we use self.now() in the "end" bound of the query).
+        #
+        # Use SENTRY_SNUBA_INFO=true while running queries in tests to know more about how data is actually queried
+        # at the clickhouse level.
+        #
+        # The solution proposed aims at solving the problem of flaky tests that occurred during CI at specific times.
+        self.store_metric(
+            org_id=self.organization.id if org_id is None else org_id,
+            project_id=self.project.id if project_id is None else project_id,
+            type=extract_entity_from_mri(name) if type is None else type,
+            name=name,
+            tags=tags,
+            timestamp=(
+                self.now()
+                - timedelta(
+                    hours=hours_before_now,
+                    minutes=minutes_before_now,
+                    # We subtract 1 second -(+1) in order to account for right non-inclusivity in the queries.
+                    #
+                    # E.g.: if we save at 10:00:00, and we have as "end" of the query that time, we must store our
+                    # value with a timestamp less than 10:00:00 so that irrespectively of the bucket we will have
+                    # the value in the query result set. This is because when we save 10:00:00 - 1 second in the db it
+                    # will be saved under different granularities as (09:59:59, 09:59:00, 09:00:00) and these are the
+                    # actual timestamps that will be compared to the bounds "start" and "end".
+                    # Supposing we store 09:59:59, and we have "start"=09:00:00 and "end"=10:00:00, and we want to query
+                    # by granularity (60 = minutes) then we look at entries with timestamp = 09:59:00 which is
+                    # >= "start" and < "end" thus all these records will be returned.
+                    # Of course this - 1 second "trick" is just to abstract away this complexity, but it can also be
+                    # avoided by being more mindful when it comes to using the "end" bound, however because we would
+                    # like our tests to be deterministic we would like to settle on this approach. This - 1 can also
+                    # be avoided by choosing specific frozen times depending on granularities and stored data but
+                    # as previously mentioned we would like to standardize the time we choose unless there are specific
+                    # cases.
+                    seconds=seconds_before_now + 1,
+                )
+            ).timestamp(),
+            value=value,
+            use_case_id=use_case_id,
+        )
+
+    def store_performance_metric(
```

Comment:
is. any chance to reuse https://github.com/getsentry/sentry/blob/20bd2d8c028dd8d409f2d86b26b7d36371f1b5da/src/sentry/testutils/cases.py#L1157-L1146 ?

Reply:
We are using it https://github.com/getsentry/sentry/blob/cb8268ba992695e21b8fd37dfa70d97ec91b4dbd/src/sentry/testutils/cases.py#L1269-L1269